lib/facility.rb

```diff
@@ -1,14 +1,78 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
+  def initialize(facilities_info)
+    @name = facilities_info[:name]
+    @address = facilities_info[:address]
+    @phone = facilities_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+  
   end
-
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    # return false unless @services.include?('Register Vehicle')
+    vehicle.registration_date = Date.today
+    @registered_vehicles << vehicle
+    vehicle
+  end
```

Comment:
We also want the facility to collect fees upon registration, so after line 19, we'd want to call `collect_fees`

Reply:
Ahhh, actually, I see what you're doing. You have a method below, `collected_fees` that will run through all the registered vehicles and calculate the total when that method is called. That certainly is a valid way to do that, however, if this is how you're doing the fees, then you don't need an instance variable of `@collected_fees` to store that data since it's being calculated dynamically whenever you call the `collected_fees` method. 
Another thing to keep in mind, is that usually variables are nouns like `collected_fees` whereas methods represent actions so we name name verbs like `collect_fees` or `calculate_fees`